spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -173,12 +173,22 @@ protected void sendMessage(Message<?> message) {
 		catch (RuntimeException e) {
 			MessageChannel errorChannel = getErrorChannel();
 			if (errorChannel != null) {
-				this.messagingTemplate.send(errorChannel, new ErrorMessage(e));
+				this.messagingTemplate.send(errorChannel, buildErrorMessage(e));
```

Comment:
Wouldn't it be preferable to delegate this to an injected `ErrorMessageStrategy` rather than invoking the overridable method below ?